docs/src/components/Layout/GlobalNav/GlobalNav.tsx

```diff
@@ -43,46 +43,41 @@ export function GlobalNav({
   // provided and the sites not using a ThemeProvider will have the needed variables added on
   const themeClass = themeableSites[currentSite] ? '' : 'use-ui-theme';
 
-  let hasSecondaryNav =
-    secondaryNavDesktop && secondaryNavMobile ? true : false;
-
-  const navLinksContainerRef = useRef<HTMLDivElement>(null);
-  const navLinksRightRef = useRef<HTMLDivElement>(null);
-
-  const allLinks = useMemo(() => {
-    return [...leftLinks, ...rightLinks];
-  }, [leftLinks, rightLinks]);
-
-  const isMobileState = useNavLinksCollision(
-    navLinksContainerRef,
-    navLinksRightRef
-  );
+  const [isCollapsed, setIsCollapsed] = useState(true);
 
   return (
     <View
-      id="main-nav"
-      className={`${styles.navbar} ${styles[themeClass]}`}
-      aria-label="Amplify Dev Center - External links to additional Amplify resources"
       as="nav"
+      className={`${styles['navbar']} ${themeClass ? styles[themeClass] : ''}`}
+      aria-label="Amplify Dev Center - External links to additional Amplify resources"
     >
-      {isMobileState ? (
-        <MobileNav
-          allLinks={allLinks}
-          hasSecondaryNav={hasSecondaryNav}
-          currentSite={currentSite}
-          secondaryNavMobile={secondaryNavMobile}
-        />
-      ) : (
-        <DesktopNav
-          currentSite={currentSite}
-          leftLinks={leftLinks}
+      <Flex className={styles['nav-links-container']}>
+        <Flex height="100%" id="left-nav" className={styles['left-nav-links']}>
+          <AmplifyNavLink
+            currentSite={currentSite}
+            isCollapsed={isCollapsed}
+            setIsCollapsed={setIsCollapsed}
+          />
+
+          <LeftNavLinks
+            isCollapsed={isCollapsed}
+            leftLinks={leftLinks}
+            currentSite={currentSite}
+          />
+        </Flex>
+        <RightNavLinks
           rightLinks={rightLinks}
-          navLinksContainerRef={navLinksContainerRef}
-          navLinksRightRef={navLinksRightRef}
-          hasSecondaryNav={hasSecondaryNav}
-          secondaryNavDesktop={secondaryNavDesktop}
+          socialLinks={socialLinks}
+          currentSite={currentSite}
+          isCollapsed={isCollapsed}
         />
-      )}
+      </Flex>
+      <View
```

Comment:
Is this needed? Why is this empty?

Reply:
We're using this div as the background overlay when the mobile nav is expanded

Reply:
Thanks, that makes sense!